tests/test_preconditions.py

```diff
@@ -20,35 +21,58 @@ def start_gprofiler(
     privileged: bool = True,
     user: int = 0,
     pid_mode: Optional[str] = "host",
+    extra_profiler_args: Optional[List[str]] = None,
 ) -> Container:
+    extra_profiler_args = extra_profiler_args or []
     return start_gprofiler_in_container_for_one_session(
         docker_client,
         gprofiler_docker_image,
         Path("/tmp"),
         Path("/tmp/collapsed"),
         [],
-        ["-d", "1"],
+        ["-d", "1"] + extra_profiler_args,
         privileged=privileged,
         user=user,
         pid_mode=pid_mode,
     )
 
 
+@contextmanager
+def run_gprofiler_container(
+    docker_client: DockerClient,
+    gprofiler_docker_image: Image,
+    privileged: bool = True,
+    user: int = 0,
+    pid_mode: Optional[str] = "host",
+    extra_profiler_args: Optional[List[str]] = None,
+) -> Container:
+    container: Container = None
+    try:
+        container = start_gprofiler(
+            docker_client, gprofiler_docker_image, privileged, user, pid_mode, extra_profiler_args
+        )
+        yield container
+    finally:
+        if container is not None:
+            container.stop()  # Fine to be run also if already stopped
+
+
 def test_mutex_taken_twice(
     docker_client: DockerClient,
     gprofiler_docker_image: Image,
 ) -> None:
     """
     Mutex can only be taken once. Second gProfiler executed should fail with the mutex already taken error.
     """
-    gprofiler1 = start_gprofiler(docker_client, gprofiler_docker_image)
-    gprofiler2 = start_gprofiler(docker_client, gprofiler_docker_image)
-
-    # exits without an error
-    assert wait_for_container(gprofiler2) == (
-        "Could not acquire gProfiler's lock. Is it already running?"
-        " Try 'sudo netstat -xp | grep gprofiler' to see which process holds the lock.\n"
-    )
+    # Run the first one continuously
```

Comment:
This is the bug then, right? The race when the first one already exits

Reply:
1. the first one might have exited before 2 have run
2. the first one didn't acquire the mutex and the second one will acquire it before (with the current solution we're waiting for log that is being logged only after the mutex successfully acquired)